flaml/nlp/autotransformers.py

```diff
@@ -0,0 +1,787 @@
+        auto_tokentoids_config = {"max_seq_length": self._max_seq_length}
+        self._tokenizer = AutoTokenizer.from_pretrained(jobid_config.pre_full, use_fast=True)
+
+        data_encoded = AutoToEncoded.from_model_and_dataset_name(data_raw,jobid_config.pre_full,jobid_config.dat[0], jobid_config.subdat, **auto_tokentoids_config)
+        self._max_seq_length = 0
+        for each_fold in data_encoded.keys():
+            self._max_seq_length = max(self._max_seq_length,
+                max([sum(data_encoded[each_fold][x]['attention_mask']) for x in range(len(data_encoded[each_fold]))]))
+        self._max_seq_length = int((self._max_seq_length + 15) / 16) * 16
+        data_encoded = AutoToEncoded.from_model_and_dataset_name(data_raw, jobid_config.pre_full, jobid_config.dat[0], jobid_config.subdat,**auto_tokentoids_config)
+
+        if jobid_config.spt == "rspt":
+            all_folds_from_source = []
+            assert "source" in resplit_portion.keys(), "Must specify the source for resplitting the dataset in" \
+            "resplit_portion, which is a list of folder names, e.g., resplit_portion = {'source': ['train']}"
+
+            source_fold_names = resplit_portion['source']
+            for each_fold_name in source_fold_names:
+                this_fold_dataset = data_encoded[each_fold_name]
+                all_folds_from_source.append(this_fold_dataset)
+
+            merged_folds_from_source = datasets.concatenate_datasets(all_folds_from_source)
+            merged_folds_from_source = merged_folds_from_source.shuffle(seed=42)
+
+            assert "train" in resplit_portion.keys() and "validation" in resplit_portion.keys() \
+                and "test" in resplit_portion.keys(), "train, validation, test must exist in resplit_portion"
+
+            for key in ["train", "validation", "test"]:
+                target_fold_start, target_fold_end = int(resplit_portion[key][0] * len(merged_folds_from_source)), \
+                        int(resplit_portion[key][1] * len(merged_folds_from_source))
+                if key == "train":
+                    self.train_dataset = merged_folds_from_source.select([x for x in range(target_fold_start, target_fold_end)]).flatten_indices()
+                elif key == "validation":
+                    self.eval_dataset = merged_folds_from_source.select([x for x in range(target_fold_start, target_fold_end)]).flatten_indices()
+                else:
+                    self.test_dataset = merged_folds_from_source.select([x for x in range(target_fold_start, target_fold_end)]).flatten_indices()
+        else:
+            self.train_dataset, self.eval_dataset, self.test_dataset = data_encoded[self._train_name], data_encoded[self._dev_name], data_encoded[
+                self._test_name]
+
+    def _set_model_config(self, checkpoint_path, per_model_config, model_config_num_labels):
+        if per_model_config and len(per_model_config) > 0:
```

Comment:
len(per_model_config) > 0 is not needed

Reply:
per_model_config is either a dict or None. The first condition checks None, the 2nd condition checks whether the dict is empty